File: 2025/SoyoungJung/week1/2025-06-08-group-anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
  const result: { [key: string]: string[] } = {};

  for (const word of strs) {
    const sorted = word.split("").sort().join("");
    if (!result[sorted]) {
      result[sorted] = [];
    }
    result[sorted].push(word);
  }

  return Object.values(result).sort();
}

File: 2025/HyeGyung/2025-06-08-Group-Anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        sorted_dict = {}
        for text in strs:
            sorted_text = ''.join(sorted(text))
            if sorted_text in sorted_dict:
                sorted_dict[sorted_text].append(text)
            else:
                sorted_dict[sorted_text] = [text]
        return list(sorted_dict.values())
